src/sagemaker/amazon/ipinsights.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+from sagemaker.amazon.amazon_estimator import AmazonAlgorithmEstimatorBase, registry
+from sagemaker.amazon.hyperparameter import Hyperparameter as hp  # noqa
+from sagemaker.amazon.validation import ge, le
+from sagemaker.predictor import RealTimePredictor, csv_serializer, json_deserializer
+from sagemaker.model import Model
+from sagemaker.session import Session
+from sagemaker.vpc_utils import VPC_CONFIG_DEFAULT
+
+
+class IPInsights(AmazonAlgorithmEstimatorBase):
+    repo_name = 'ipinsights'
+    repo_version = 1
+    MINI_BATCH_SIZE = 10000
+
+    num_entity_vectors = hp('num_entity_vectors', (ge(1), le(250000000)), 'An integer in [1, 250000000]', int)
+    vector_dim = hp('vector_dim', (ge(4), le(4096)), 'An integer in [4, 4096]', int)
+
+    batch_metrics_publish_interval = hp('batch_metrics_publish_interval', (ge(1)), 'An integer greater than 0', int)
+    epochs = hp('epochs', (ge(1)), 'An integer greater than 0', int)
+    learning_rate = hp('learning_rate', (ge(1e-6), le(10.0)), 'A float in [1e-6, 10.0]', float)
+    num_ip_encoder_layers = hp('num_ip_encoder_layers', (ge(0), le(100)), 'An integer in [0, 100]', int)
+    random_negative_sampling_rate = hp('random_negative_sampling_rate', (ge(0), le(500)), 'An integer in [0, 500]', int)
+    shuffled_negative_sampling_rate = hp('shuffled_negative_sampling_rate', (ge(0), le(500)), 'An integer in [0, 500]',
+                                         int)
+    weight_decay = hp('weight_decay', (ge(0.0), le(10.0)), 'A float in [0.0, 10.0]', float)
+
+    def __init__(self, role, train_instance_count, train_instance_type, num_entity_vectors, vector_dim,
+                 batch_metrics_publish_interval=None, epochs=None, learning_rate=None,
+                 num_ip_encoder_layers=None, random_negative_sampling_rate=None,
+                 shuffled_negative_sampling_rate=None, weight_decay=None, **kwargs):
+        """IP Insights is :class:`Estimator` used for unsupervised learning.
```

Comment:
nitpick: change this to either

```
``IPInsights`` is a :class:`Estimator` used for unsupervised learning.
```

or something like

```
This estimator is for IP Insights, an algorithm for unsupervised learning.
```

Reply:
Updated to:

```
This estimator is for IP Insights, an unsupervised algorithm that learns usage patterns of IP addresses.
```